zm/zm_convr.F90

```diff
@@ -0,0 +1,3108 @@
+            dqsdtp(i,k) = qstp(i,k)* (1._kind_phys+qstp(i,k)/eps1)*eps1*rl/(rd*tp(i,k)**2)
+!
+! dtpdt is the parcel temperature change due to change of
+! subcloud layer properties during convection.
+!
+            dtpdt(i,k) = tp(i,k)/ (1._kind_phys+rl/cp* (dqsdtp(i,k)-qstp(i,k)/tp(i,k)))* &
+                        (dtbdt(i)/t(i,mx(i))+rl/cp* (dqbdt(i)/tl(i)-q(i,mx(i))/ &
+                         tl(i)**2*dtldt(i)))
+!
+! dboydt is the integrand of cape change.
+!
+            dboydt(i,k) = ((dtpdt(i,k)/tp(i,k)+1._kind_phys/(1._kind_phys+1.608_kind_phys*qstp(i,k)-q(i,mx(i)))* &
+                          (1.608_kind_phys * dqsdtp(i,k) * dtpdt(i,k) -dqbdt(i))) - (dtmdt(i,k)/t(i,k)+0.608_kind_phys/ &
+                          (1._kind_phys+0.608_kind_phys*q(i,k))*dqmdt(i,k)))*grav*thetavp(i,k)/thetavm(i,k)
+         end if
+      end do
+   end do
+!
+   do k = msg + 1,pver
+      do i = il1g,il2g
+         if (k > lcl(i) .and. k < mx(i)) then
+            thetavp(i,k) = tp(i,k)* (1000._kind_phys/p(i,k))** (rd/cp)*(1._kind_phys+0.608_kind_phys*q(i,mx(i)))
+            thetavm(i,k) = t(i,k)* (1000._kind_phys/p(i,k))** (rd/cp)*(1._kind_phys+0.608_kind_phys*q(i,k))
+!
+! dboydt is the integrand of cape change.
+!
+            dboydt(i,k) = (dtbdt(i)/t(i,mx(i))+0.608_kind_phys/ (1._kind_phys+0.608_kind_phys*q(i,mx(i)))*dqbdt(i)- &
+                          dtmdt(i,k)/t(i,k)-0.608_kind_phys/ (1._kind_phys+0.608_kind_phys*q(i,k))*dqmdt(i,k))* &
+                          grav*thetavp(i,k)/thetavm(i,k)
+         end if
+      end do
+   end do
+
+!
+! buoyant energy change is set to 2/3*excess cape per 3 hours
+!
+   dadt(il1g:il2g)  = 0._kind_phys
+   kmin = minval(lel(il1g:il2g))
+   kmax = maxval(mx(il1g:il2g)) - 1
+   do k = kmin, kmax
+      do i = il1g,il2g
+         if ( k >= lel(i) .and. k <= mx(i) - 1) then
+            dadt(i) = dadt(i) + dboydt(i,k)* (zf(i,k)-zf(i,k+1))
+         endif
+      end do
+   end do
+   do i = il1g,il2g
+      dltaa = -1._kind_phys* (cape(i)-capelmt)
+      if (dadt(i) /= 0._kind_phys) mb(i) = max(dltaa/tau/dadt(i),0._kind_phys)
+   end do
+!
+   return
+end subroutine closure
+
+subroutine q1q2_pjr(pcols   ,pver    ,latice  ,&
+                    dqdt    ,dsdt    ,q       ,qs      ,qu      , &
+                    su      ,du      ,qhat    ,shat    ,dp      , &
+                    mu      ,md      ,sd      ,qd      ,ql      , &
+                    dsubcld ,jt      ,mx      ,il1g    ,il2g    , &
+                    cp      ,rl      ,msg     ,          &
+                    dl      ,evp     ,cu)
+
+   implicit none
+
+!-----------------------------------------------------------------------
+!
+! Purpose:
+! <Say what the routine does>
+!
+! Method:
+! <Describe the algorithm(s) used in the routine.>
+! <Also include any applicable external references.>
+!
+! Author: phil rasch dec 19 1995
+!
+!-----------------------------------------------------------------------
```

Comment:
Not sure if its possible but if we could find someone that can document this section as well, that might be worthwhile.